test/schema/cgmSettingsTest.js

```diff
@@ -0,0 +1,141 @@
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+ /* global describe, before, beforeEach, it, after */
+
+'use strict';
+
+var _ = require('lodash');
+var expect = require('salinity').expect;
+
+var helper = require('./schemaTestHelper.js');
+
+var goodObject = {
+  type: 'cgmSettings',
+  time: '2015-06-16T13:09:10.000Z',
+  timezoneOffset: 120,
+  deviceId: 'test',
+  uploadId: 'test',
+  transmitterId: 'test',
+  units: 'mg/dL',
+  "highAlerts": {
+      "enabled": true,
+      "level": 8.3261219865683,
+      "snooze": 0
+  },
+  "lowAlerts": {
+      "enabled": true,
+      "level": 3.6079861941795968,
+      "snooze": 0
+  },
+  "outOfRangeAlerts": {
+      "enabled": false,
+      "snooze": 1800000
+  },
+  "rateOfChangeAlerts": {
+      "fallRate": {
+          "enabled": false,
+          "rate": -0.16652243973136602
+      },
+      "riseRate": {
+          "enabled": false,
+          "rate": 0.16652243973136602
+      }
+  },
+  _groupId: 'g'
+};
+
+describe('schema/cgmSettings.js', function () {
+  describe('transmitterId', function () {
+    helper.rejectIfAbsent(goodObject, 'transmitterId');
+    helper.expectStringField(goodObject, 'transmitterId');
+  });
+  describe('units', function () {
+    helper.rejectIfAbsent(goodObject, 'units');
+    helper.expectStringField(goodObject, 'units');
+    helper.expectFieldIn(goodObject, 'units',
+      ['mmol/L', 'mmol/l', 'mg/dL', 'mg/dl'],
+      ['mmol/L', 'mmol/L', 'mg/dL', 'mg/dL']);
+  });
+  describe('highAlerts', function () {
+    helper.rejectIfAbsent(goodObject, 'highAlerts');
+    helper.expectObjectField(goodObject, 'highAlerts');
+
+    it('fields', function(done){
+      var localGood = _.cloneDeep(goodObject);
+      helper.rejectIfAbsent(localGood.lowAlerts, 'enabled');
+      helper.expectObjectField(localGood.lowAlerts, 'enabled');
+      helper.rejectIfAbsent(localGood.lowAlerts, 'level');
+      helper.expectNumericalField(localGood.lowAlerts, 'level');
+      helper.rejectIfAbsent(localGood.lowAlerts, 'snooze');
+      helper.expectNumericalField(localGood.lowAlerts, 'snooze');
+      done();
+    });
+  });
+  describe('lowAlerts', function () {
+    helper.rejectIfAbsent(goodObject, 'lowAlerts');
+    helper.expectObjectField(goodObject, 'lowAlerts');
+
+    it('fields', function(done){
+      var localGood = _.cloneDeep(goodObject);
+      helper.rejectIfAbsent(localGood.lowAlerts, 'enabled');
+      helper.expectObjectField(localGood.lowAlerts, 'enabled');
+      helper.rejectIfAbsent(localGood.lowAlerts, 'level');
+      helper.expectNumericalField(localGood.lowAlerts, 'level');
+      helper.rejectIfAbsent(localGood.lowAlerts, 'snooze');
+      helper.expectNumericalField(localGood.lowAlerts, 'snooze');
+      done();
+    });
+  });
+  describe('outOfRangeAlerts', function () {
```

Comment:
Would you mind adding the note about this being an optional field? Helpful for those reading the tests later, I think.